src/mbgl/actor/scheduler.cpp

```diff
@@ -1,22 +1,50 @@
 #include <mbgl/actor/scheduler.hpp>
-#include <mbgl/util/thread_local.hpp>
 #include <mbgl/util/thread_pool.hpp>
 
+#undef THREAD_LOCAL_UNSUPPORTED
+#ifdef __APPLE__
+# include <TargetConditionals.h>
+# include <Availability.h>
+# if defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED < __IPHONE_9_0
+#   define THREAD_LOCAL_UNSUPPORTED = 1
+# endif
+# if defined(__i386__) && defined(TARGET_OS_IPHONE)
+#   define THREAD_LOCAL_UNSUPPORTED = 1
+# endif
+#endif
+
+#ifdef THREAD_LOCAL_UNSUPPORTED
+#include <mbgl/util/thread_local.hpp>
+#endif
```

Comment:
Let's unify this code in thread_local.hpp to avoid having to repeat this `#ifdef` tree multiple times.